ui/src/message_popup/mobile_prototype/texting_interface.jsx

```diff
@@ -0,0 +1,184 @@
+/* @flow weak */
+import React from 'react';
+import _ from 'lodash';
+import VelocityComponent from 'velocity-react/velocity-component';
+import 'velocity-animate/velocity.ui';
+
+import Divider from 'material-ui/Divider';
+import TextField from 'material-ui/TextField';
+import RaisedButton from 'material-ui/RaisedButton';
+
+import {StudentMessage, UserMessage, InfoMessage} from './message_components.jsx';
+
+export const TextBody = React.createClass({
+  displayName: 'TextBody',
+
+  propTypes: {
+    question: React.PropTypes.object.isRequired,
+    messages: React.PropTypes.array.isRequired,
+    onOpenStudentDialog: React.PropTypes.func.isRequired,
+    onOpenInfoDialog: React.PropTypes.func.isRequired
+  },
+  
+  render(){
```

Comment:
Nice!  This is great factoring to pull out the components of the messages here, it makes this clear and direct.